### ISSUE TITLE
Wrong failed query on numeric query

### ISSUE THREAD

**Issue by JorgeGarciaIrazabal** [sodadata/soda-core#1800]
Soda report a wrong query to retrieve the failed records over a numeric query.

How to reproduce:
```
import duckdb
import pandas as pd
from soda.sampler.sampler import Sampler
from soda.scan import Scan

duckdb_conn = duckdb.connect(database=":memory:")


class QuerySampler(Sampler):
    def store_sample(self, sample_context):
        print("query:", sample_context.query)
        output = duckdb_conn.execute(sample_context.query).df()
        print("values:", set(output.value))


df = pd.DataFrame({"value": [1, 2, 3, 4, 5, 6]})
duckdb_conn.register("my_table", df)

scan = Scan()
scan.add_configuration_yaml_str(
    f"""
data_source duckdb_db:
  type: duckdb
  database: duckdb
"""
)


scan.set_data_source_name("duckdb_db")
scan.add_duckdb_connection(duckdb_conn, data_source_name="duckdb_db")
scan.add_sodacl_yaml_str(
    """
checks for my_table:
- invalid_count(value) = 0:
   valid max: 4
   valid min: 2
   """
)
scan.sampler = QuerySampler()
scan.execute()

```

Current output (from `QuerySampler`)
```
SELECT * FROM my_table 
 WHERE NOT value IS NULL AND NOT value >= 2.0 AND value <= 4.0 
 LIMIT 100
values: {1}

```

Expected output:

```
query: SELECT * FROM my_table 
 WHERE NOT value IS NULL AND NOT (value >= 4.0 AND value <= 3.0)
 LIMIT 100
values: {1, 5, 6}
```

I was able to reproduce this with Postgressql too

**Comment by jmarien** [comment]
SODA-1441

**Comment by JorgeGarciaIrazabal** [comment]
I think the problem is here: https://github.com/sodadata/soda-core/blob/main/soda/core/soda/execution/metric/numeric_query_metric.py#L312

I can create a PR today if that helps.

**Comment by JorgeGarciaIrazabal** [comment]
fixed on #1811 closing this issue
